From e1edfeaf082f409a4fda2b3f1a6de2b2f53c48f5 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Fri, 30 Jan 2015 13:41:38 -0500
Subject: [PATCH 037/315] manager: drop wait-for-go

GDM used to have this concept called "wait-for-go" where it would
stall at boot up and wait for a go signal before proceeding. This
feature never got fully reimplemented after the rewrite in '22 and
and it's not used anyway.

Drop it.

https://bugzilla.gnome.org/show_bug.cgi?id=744764
---
 daemon/gdm-manager.c | 28 ++--------------------------
 daemon/gdm-manager.h |  2 --
 2 files changed, 2 insertions(+), 28 deletions(-)

diff --git a/daemon/gdm-manager.c b/daemon/gdm-manager.c
index fa3563e6..ab07e139 100644
--- a/daemon/gdm-manager.c
+++ b/daemon/gdm-manager.c
@@ -78,7 +78,6 @@ struct GdmManagerPrivate
         GCancellable           *cancellable;
 
         gboolean                started;
-        gboolean                wait_for_go;
         gboolean                show_local_greeter;
 
         GDBusProxy               *bus_proxy;
@@ -2065,13 +2064,13 @@ gdm_manager_start (GdmManager *manager)
 {
         g_debug ("GdmManager: GDM starting to manage displays");
 
-        if (! manager->priv->wait_for_go && (!manager->priv->xdmcp_enabled || manager->priv->show_local_greeter)) {
+        if (!manager->priv->xdmcp_enabled || manager->priv->show_local_greeter) {
                 gdm_display_factory_start (GDM_DISPLAY_FACTORY (manager->priv->local_factory));
         }
 
 #ifdef HAVE_LIBXDMCP
         /* Accept remote connections */
-        if (manager->priv->xdmcp_enabled && ! manager->priv->wait_for_go) {
+        if (manager->priv->xdmcp_enabled) {
                 if (manager->priv->xdmcp_factory != NULL) {
                         g_debug ("GdmManager: Accepting XDMCP connections...");
                         gdm_display_factory_start (GDM_DISPLAY_FACTORY (manager->priv->xdmcp_factory));
@@ -2082,29 +2081,6 @@ gdm_manager_start (GdmManager *manager)
         manager->priv->started = TRUE;
 }
 
-void
-gdm_manager_set_wait_for_go (GdmManager *manager,
-                             gboolean    wait_for_go)
-{
-        if (manager->priv->wait_for_go != wait_for_go) {
-                manager->priv->wait_for_go = wait_for_go;
-
-                if (! wait_for_go) {
-                        /* we got a go */
-                        if (!manager->priv->xdmcp_enabled || manager->priv->show_local_greeter) {
-                                gdm_display_factory_start (GDM_DISPLAY_FACTORY (manager->priv->local_factory));
-                        }
-
-#ifdef HAVE_LIBXDMCP
-                        if (manager->priv->xdmcp_enabled && manager->priv->xdmcp_factory != NULL) {
-                                g_debug ("GdmManager: Accepting XDMCP connections...");
-                                gdm_display_factory_start (GDM_DISPLAY_FACTORY (manager->priv->xdmcp_factory));
-                        }
-#endif
-                }
-        }
-}
-
 static gboolean
 register_manager (GdmManager *manager)
 {
diff --git a/daemon/gdm-manager.h b/daemon/gdm-manager.h
index 4482bdd1..41c68a7a 100644
--- a/daemon/gdm-manager.h
+++ b/daemon/gdm-manager.h
@@ -66,8 +66,6 @@ GType               gdm_manager_get_type                       (void);
 GdmManager *        gdm_manager_new                            (void);
 void                gdm_manager_start                          (GdmManager *manager);
 void                gdm_manager_stop                           (GdmManager *manager);
-void                gdm_manager_set_wait_for_go                (GdmManager *manager,
-                                                                gboolean    wait_for_go);
 
 void                gdm_manager_set_xdmcp_enabled              (GdmManager *manager,
                                                                 gboolean    enabled);
-- 
2.15.1

